packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/index.tsx

```diff
@@ -35,16 +35,22 @@ export interface SelectProviderDialogProps {}
 export function SelectProviderDialog(props: SelectProviderDialogProps) {
     const { t } = useI18N()
     const { classes } = useStyles()
-
+    const [onlyEvm, setOnlyEvm] = useState(false)
     // #region remote controlled dialog logic
-    const { open, closeDialog } = useRemoteControlledDialog(WalletMessages.events.selectProviderDialogUpdated)
     // #endregion
-
+    const { open, closeDialog } = useRemoteControlledDialog(WalletMessages.events.selectProviderDialogUpdated)
     // #region native app
     useEffect(() => {
         if (!open) return
+        WalletMessages.events.selectProviderDialogUpdated.on((ev?) => {
+            if ((ev as { onlyEvm: boolean }).onlyEvm) {
```

Comment:
You can rather change the event type than force typecasting.

Reply:
Recommend to add a local state `pluginID` and set it from the event. With it, you can filter networks and pass them down to `< PluginProviderRender  />`.